File: apps/backend/internal/module/metadata/api/master_detail.go
package api

import (
	"context"
	"metadata-platform/internal/module/metadata/service"
	"metadata-platform/internal/utils"

	"github.com/cloudwego/hertz/pkg/app"
	"github.com/cloudwego/hertz/pkg/protocol/consts"
)

// MasterDetailHandler 主子表处理器
type MasterDetailHandler struct {
	mdService service.MasterDetailService
}

// NewMasterDetailHandler 创建主子表处理器实例
func NewMasterDetailHandler(mdService service.MasterDetailService) *MasterDetailHandler {
	return &MasterDetailHandler{mdService: mdService}
}

// CreateMasterDetail 创建主子表数据
func (h *MasterDetailHandler) CreateMasterDetail(c context.Context, ctx *app.RequestContext) {
	masterModelID := ctx.Param("master")
	detailModelID := ctx.Param("detail")

	var payload map[string]any
	if err := ctx.BindJSON(&payload); err != nil {
		utils.ErrorResponse(ctx, consts.StatusBadRequest, "Invalid JSON payload")
		return
	}

	if err := h.mdService.CreateMasterDetail(c, masterModelID, detailModelID, payload); err != nil {
		utils.ErrorResponse(ctx, consts.StatusInternalServerError, err.Error())
		return
	}
	utils.SuccessResponse(ctx, "Created successfully")
}

File: apps/frontend/src/api/master-detail.ts
/**
 * 主子表事务API服务
 */
import request from '@/utils/request'

/**
 * 创建主子表事务
 * @param masterModelName 主表模型名
 * @param detailModelName 子表模型名
 * @param data 事务提交数据 { master: {...}, details: [...] }
 * @returns 创建结果
 */
export const createMasterDetail = async (masterModelName: string, detailModelName: string, data: any): Promise<any> => {
    return request({
        url: `/api/metadata/master-detail/${masterModelName}/${detailModelName}`,
        method: 'post',
        data
    })
}
